integration/test_timescaledb.py

```diff
@@ -0,0 +1,267 @@
+    """Fixture for extended type testing"""
+    timestamp_precision, varchar_size = request.param
+    test_type = "extended"
+    table_name = f"test_table_{test_type}_{timestamp_precision}"
+
+    # Setup database connection
+    if dsn:
+        connection_params = _parse_dsn_for_connection_params(dsn)
+        conn_host = connection_params["host"]
+        conn_port = connection_params["port"]
+        conn_database = connection_params["database"]
+        conn_user = connection_params["user"]
+        conn_password = connection_params["password"]
+        connection = psycopg2.connect(dsn)
+
+    connection.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+    test_connection = connection
+    test_cursor = test_connection.cursor()
+
+    # Create TimescaleDB extension
+    test_cursor.execute("CREATE EXTENSION IF NOT EXISTS timescaledb;")
+
+    # Drop and create extended table
+    with contextlib.suppress(psycopg2.Error):
+        test_cursor.execute(f"DROP TABLE IF EXISTS {table_name};")
+    test_cursor.execute(
+        f"""
+        CREATE TABLE {table_name} (
+            time TIMESTAMPTZ NOT NULL,
+            binary_col {DATATYPE_MAPPING["BINARY"]},
+            bool_col {DATATYPE_MAPPING["BOOL"]},
+            double_col {DATATYPE_MAPPING["DOUBLE"]},
+            float_col {DATATYPE_MAPPING["FLOAT"]},
+            int_col {DATATYPE_MAPPING["INT"]},
+            timestamp_col {DATATYPE_MAPPING["TIMESTAMP"]},
+            nchar_col {DATATYPE_MAPPING["NCHAR"]}(10),
+            varchar_col {DATATYPE_MAPPING["VARCHAR"]}({varchar_size})
+        );
+    """
+    )
+
+    test_cursor.execute(f"SELECT create_hypertable('{table_name}', 'time');")
+    test_cursor.close()
+
+    columns_config = [
+        "binary_col",
+        "bool_col",
+        "double_col",
+        "float_col",
+        "int_col",
+        "timestamp_col",
+        "nchar_col",
+        "varchar_col",
+    ]
+
+    yield (
+        test_connection,
+        conn_host,
+        conn_port,
+        conn_database,
+        conn_user,
+        conn_password,
+        table_name,
+        dsn,
+        timestamp_precision,
+        columns_config,
+    )
+
+    # Cleanup
+    try:
+        cleanup_cursor = test_connection.cursor()
+        cleanup_cursor.execute(f"DROP TABLE IF EXISTS {table_name} CASCADE;")
+        cleanup_cursor.close()
+    except Exception as e:
+        print(f"Cleanup error: {e}")
+    test_connection.close()
+
+
+def test_timescaledb_all_types_and_precision(timescaledb):
+    """Comprehensive test for all data types with millisecond and microsecond precision"""
+    (connection, host, port, db_name, user, password, table_name, dsn_url, timestamp_precision, columns_config) = (
+        timescaledb
+    )
+
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    if timestamp_precision in ["milliseconds", "microseconds"]:
+        time_format = "%d/%m/%y %H:%M:%S.%f UTC%z"
+
+    controller = build_flow(
+        [
+            SyncEmitSource(),
```

Comment:
Need to test with `AsyncEmitSource`, as I suspect that it will fail (as per my previous comment regarding `run_until_complete`.

Reply:
Done